process/handlers.lua

```diff
@@ -116,28 +164,80 @@ function handlers.after(handleName)
 end
 
 function handlers.remove(name)
-  assert(type(name) == 'string', 'name MUST be string')
-  if #handlers.list == 1 and handlers.list[1].name == name then
-    handlers.list = {}
-    
+  assert(type(name) == 'string', 'Handler name MUST be a string')
+  
+  local idx = findIndexByProp(handlers.list, "name", name)
+  if idx then
+    table.remove(handlers.list, idx)
+  else
+    error('Handler not found')
   end
+end
 
-  local idx = findIndexByProp(handlers.list, "name", name)
-  table.remove(handlers.list, idx)
-  
+function handlers.matchesPattern(msg, pattern)
+  if type(pattern) == 'function' then
+    return pattern(msg)
+  -- If the pattern is a table, step through every key/value pair in the pattern and check if the msg matches
+  -- Supported match types:
+  --   - Exact string match
+  --   - Lua gmatch string
+  --   - '_' (wildcard: Message has tag, but can be any value)
+  --   - Function execution on the tag, optionally using the msg as the second argument
+  end
+  if type(pattern) == 'table' then
+    for key, patternMatchSpec in pairs(pattern) do
+      local matched = false
+      -- If the key is not in the message, then it does not match
+      if(not msg[key]) then
+        return false
+      end
+      -- if the patternMatchSpec is a wildcard, then it always matches
+      if patternMatchSpec == '_' then
+        matched = true
+      end
+      -- if the patternMatchSpec is a function, then it is executed on the tag value
+      if type(patternMatchSpec) == "function" then
+        if patternMatchSpec(msg[key], msg) then
+          matched = true
+        else
+          return false
```

Comment:
Same behavior as line 192 -- this will short circuit checking any other `MatchSpec` on the first `MatchSpec` function to not return `truthy`.